### ISSUE TITLE
output_files not working properly

### ISSUE THREAD

**Issue by Xiaohequ** [bugy/script-server#542]
I'm very intereasted in this projet. I'm so happy to see someone else have the same idea as me. This projet seems very well maintained, great job.

I have seen 3 bugs in latest version 1.17.0

1. script need pause (read -p) in order to make download link appear
2. ##any_path# not working ( * did work)
3. output_files "type": "inline-image" not show up

### script need pause (read -p) for download link to appear
![with_pause](https://user-images.githubusercontent.com/6079782/159157012-b0f5491c-13e5-4060-99c4-19374faecdd4.jpg)
![without_pause](https://user-images.githubusercontent.com/6079782/159157013-995e6f26-eb8e-4782-9e97-9683d5fb18b8.jpg)

### ##any_path# not working
working: "output_files " : ["/home/ubuntu/tools/script-server/output/download_file/*"]
did not working: "output_files " : ["/home/ubuntu/tools/script-server/output/download_file/##any_path#"]

### "type": "inline-image" not show up
the link was received on the browser side, but no link or image are show up
![inline_image_not_show](https://user-images.githubusercontent.com/6079782/159157380-8d768e20-3aa8-4991-ad59-2a2b8c17c667.jpg)

my config file here
[config_files.zip](https://github.com/bugy/script-server/files/8310973/config_files.zip)

**Comment by bugy** [comment]
Hi @Xiaohequ , I'm not 100% about your cases, but i believe everything works as expected:
1. Links are generated after a script is finished. Not during an execution
2. ##any_path# (as all other regular expression parsing) is used only for log parsing. I.e. it will find only the files, which were logged by a script

**Comment by Xiaohequ** [comment]
I see. Thank you for the quick response.

**Comment by Xiaohequ** [comment]
I have a question. Is `read -p "Press enter to start writing to the file (~/$filename)"` are needed for the download link to works / appear ? because i  can't manage to make the sample script (write_file.sh) work without this ligne. Did I miss something ?

modified conf file:
[write_file_without_read.zip](https://github.com/bugy/script-server/files/8311568/write_file_without_read.zip)

**Comment by bugy** [comment]
Yes, exactly. For this sample I'm using a regular expression to parse
script output
And if there is no pathes in the output, then script server doesn't have
anything for downloading

Kevin QUX ***@***.***> schrieb am So., 20. März 2022, 16:41:

**Comment by Xiaohequ** [comment]
Ok. But when I replace it by `echo "Press enter to start writing to the file (~/$filename)"`, it's still not working. Did I do something wrong ?

**Comment by bugy** [comment]
@Xiaohequ I took your configs, replaced `read` with `echo` and it still worked for me
![image](https://user-images.githubusercontent.com/1275813/159458741-f4c15117-821d-47e1-9609-c1781e94bc74.png)

**Comment by Xiaohequ** [comment]
Ok. That's odd. I'm going to test this on a different os and with different version of python. I will let you know if i found something. Thank you for looking.

**Comment by Xiaohequ** [comment]
In passing month, I have tested the script server on other système/hardware (debian, ubuntu with python 3.6). i had the same issue (not prompted to download generated file after multiple execution).
But today, when I tested it with docker image. It works great.
Then I decide to retry on my previous unsuccessful machine. It works without any issue. 
Maybe I was unlucky.
So it's all good for me. Thanks again. Have a nice day.
